.travis.yml

```diff
@@ -1,76 +1,100 @@
+    rm /etc/ros/rosdep/sources.list.d/20-default.list && 
+    rosdep init && 
+    rosdep update && 
+    mkdir -p ~/catkin_ws/src && 
+    cd ~/catkin_ws/src && 
+    apt-get update && 
+    apt-get install -y 
+        python-catkin-pkg 
+        python-rosdep 
+        python-wstool 
+        python-catkin-tools 
+        ros-kinetic-catkin 
+        ros-kinetic-rostest 
+        wget 
+        curl && 
+    source /opt/ros/kinetic/setup.bash && 
+    catkin_init_workspace && 
+    cd ~/catkin_ws && 
+    catkin build && 
+    source devel/setup.bash && 
+    cd ~/catkin_ws/src && 
+    ln -s /travis ."
 
 # Install all dependencies, using wstool first and rosdep second.
 # wstool looks for a ROSINSTALL_FILE defined in the environment variables.
 before_script:
-  # source dependencies: install using wstool.
-  - cd ~/catkin_ws/src
-  - wstool init
-  - if [[ -f $ROSINSTALL_FILE ]] ; then wstool merge $ROSINSTALL_FILE ; fi
-  - wstool up
-  # package depdencies: install using rosdep.
-  - cd ~/catkin_ws
-  - rosdep install -y --from-paths src --ignore-src --rosdistro $ROS_DISTRO
+  - docker exec -t alica-test bash -c "
+    cd ~/catkin_ws/src && 
+    wstool init && 
+    if [[ -f /travis/dependencies.rosinstall ]] ; then 
+        sed -i -e \"s/rr-devel/$BRANCH/g\" /travis/dependencies.rosinstall; 
+        if wstool merge -y /travis/dependencies.rosinstall ; then 
+            sed -i -e \"s/$BRANCH/rr-devel/g\" /travis/dependencies.rosinstall; 
+            wstool merge -y /travis/dependencies.rosinstall; 
+        fi; 
+    fi && 
+    wstool up && 
+    cd ~/catkin_ws && 
+    rosdep install -y --from-paths src --ignore-src --rosdistro kinetic"
 
 # Compile and test (mark the build as failed if any step fails). If the
 # CATKIN_OPTIONS file exists, use it as an argument to catkin_make, for example
 # to blacklist certain packages.
 #
+# If the current branch is coverity_scan, trigger a coverity analysis.
+#
 # NOTE on testing: `catkin_make run_tests` will show the output of the tests
 # (gtest, nosetest, etc..) but always returns 0 (success) even if a test
 # fails. Running `catkin_test_results` aggregates all the results and returns
 # non-zero when a test fails (which notifies Travis the build failed).
 script:
-  - source /opt/ros/$ROS_DISTRO/setup.bash
-  - cd ~/catkin_ws
-  - catkin build --no-status $( [ -f $CATKIN_OPTIONS ] && cat $CATKIN_OPTIONS )
-  # Run the tests, ensuring the path is set correctly.
-  - source devel/setup.bash
-  - catkin run_tests && catkin_test_results
+  - docker exec -t alica-test bash -c "
+    source /opt/ros/kinetic/setup.bash && 
+    cd ~/catkin_ws && 
+    catkin build --no-status && 
+    source devel/setup.bash && 
+    catkin run_tests --no-status && catkin_test_results && 
+    if [[ \"$BRANCH\" = \"coverity_scan\" ]] ; then 
+        wget -qO- https://scan.coverity.com/download/cxx/linux64 
+            --post-data \"token=o6qYj0vj-t-K1PKQj5ugyg&project=rapyuta-robotics/alica\" | 
```

Comment:
why alica in alica-supplementary?

Reply:
please make sure this token is valid on the general rapyuta account

Reply:
Believe it or not, alica depends on supplementary, and supplementary depends on alica. It's dumb, and eventually we will want to completely decouple them.

Furthermore, the token will be valid even after changing the email associated with the Coverity account.